exonum-java-binding-core/rust/src/storage/proof_map_index.rs

```diff
@@ -163,15 +174,91 @@ pub extern "system" fn Java_com_exonum_binding_storage_indices_ProofMapIndexProx
     let res = panic::catch_unwind(|| {
         let key = convert_to_key(&env, key)?;
         env.ensure_local_capacity(512)?;
```

Comment:
It probably has to be _f(num(branches) + num(leaves))_? Or we have to delete local refs as we add them to the _java_ array (i.e., do not store them in an intermediate vector.